src/components/engagement_edit_modals/user_edit_fields.tsx

```diff
@@ -76,125 +90,187 @@ export const UserEditFields = ({
                 <TableBody />
               </Table>
               <Grid hasGutter>
-                <Form isHorizontal>
+                <Form>
                   <GridItem>
                     {users.map((value: any, index: any) => {
                       const isUserDeleted =
                         deletedUsers.indexOf(users[index].email) > -1;
+                      if(validEmail === null || validName === null || validLastName === null || validRole === null) {
+                        setValidEmail(validateEmail(value.email));
```

Comment:
Why does each field need to be validated when only one field is updated?

Reply:
This is when the form load for the first time, it runs one time only. So each row would do their initial validation even before any changes. 
Foreach changes in any field, we want to make sure the other fields are getting validated, and only if all of them are correct we let the user save the input.